target/mips/cp0_helper.c

```diff
@@ -1486,21 +1588,25 @@ void helper_mtc0_taglo(CPUMIPSState *env, target_ulong arg1)
     } else {
         env->CP0_TagLo = arg1 & 0xFFFFFCF6;
     }
+    log_instr_cop0_update(env, CP0_REGISTER_28, 1, env->CP0_TagLo);
 }
 
 void helper_mtc0_datalo(CPUMIPSState *env, target_ulong arg1)
 {
     env->CP0_DataLo = arg1; /* XXX */
+    log_instr_cop0_update(env, CP0_REGISTER_28, 0, arg1);
 }
 
 void helper_mtc0_taghi(CPUMIPSState *env, target_ulong arg1)
 {
     env->CP0_TagHi = arg1; /* XXX */
+    log_instr_cop0_update(env, CP0_REGISTER_29, 1, arg1);
```

Comment:
Having to add all these logs manually is not ideal, but I guess there isn't really any other solution that results in a smaller diff.

Reply:
I couldn't think of a better way, since they are modified in various different helpers.
The alternative is to revert back to the previous way of scanning all the registers looking for changes. In my opinion that approach is worse as we lose the intent of the current instruction and we could not detect updates that rewrite the same value to the register, this helps having consistent traces without random "holes". Hopefully those helpers do not change a lot and cause many conflicts?

Reply:
I doubt there will be many future changes to the MIPS target so this should be perfectly fine.

Reply:
I doubt there will be many future changes to the MIPS target so this should be perfectly fine.